### DISCUSSION TITLE
Office Hours Agenda Topics

### DISCUSSION THREAD

**Discussion by jhaack** [VOLTTRON/volttron-core#160] [General]
Eclipse VOLTTRON Community,

Please use this space for suggestions of Office Hours topics and to sign up to present research. This could be 

- Use cases
- Features of the platform
- Research in the community
- Collaboration opportunities
- Upcoming projects
- New technologies of interest
- Updates from working groups

Please reply with your suggestions and any topics you'd like to lead discussion on. 

Thanks!

**Comment by jkempf42** [comment]
Hi,

Probably this should be low priority because it is message bus related and we will hopefully meet next week on that. 

The superclass API Craig proposed has four messages: add_group()/remove_group(), add_user_to_group()/remove_user_from_group() that I think are designed to add identities with credentials, which I assume include both auth and authz. Problem is this is very much tied to the Linux user/group identity and access management (IAM) model. If you want to use another model for IAM, for example Spiffie, you would have to map it into the Linux user/group model.  Can we discuss an API that is more agnostic about the underlying IAM model?

Thanx.

          jak

**Reply by craigpnnl** [reply]
For more context I include an updated diagram which includes the AuthService.  For a Zmq based system including all of the roles etc for writing to the auth file etc (which we will talk about) the following is a good starting point I think.   The point is the AuthService will utilize a minimal interface for doing the authentication on a resource.  
![image](https://user-images.githubusercontent.com/3979063/221255626-54f8c676-782c-4e36-89d5-9a1fa8c88bcf.png)

**Reply by craigpnnl** [reply]
This is the current 8.0 implementation that we have.  The model up above is not as comprehensive as of now.
![image](https://user-images.githubusercontent.com/3979063/221265593-900000aa-bbc8-424e-abac-90d5d409900e.png)

**Comment by Greg-I4UG** [comment]
As far as software methods/tools used for connectivity this is what I think of as minimum requirements:

- Has a C code API/ABI
- Capable of managing asynchronous socket communications using as few threads as possible (like two).
- Has methods for determining the IP/URL of a connection.
- Facilities for OpenSSL.

Greg
